### ISSUE TITLE
iomad_commerce block UI text needs updating to avoid potential database errors

### ISSUE THREAD

**Issue by OliverOPI** [iomad/iomad#1868]
TLDR: The ecommerce block needs a minor UI edit so as to avoid human error.

**Problem**: The "name" field has a maxchar of 50, but the UI does not specify this. This can result in the Moodle database throwing an error

**Affected URL**: /blocks/iomad_commerce/edit_course_shopsettings_form.php?shopsettingsid=4&default=0

**Solution**: Change the UI text to "Name (max 50 chars)" so that the end user is warned that there are character length restrictions.

**Description**:
If an admin user makes an ecommerce product title that is longer than 50 characters, it will result in Moodle throwing a database error.

![database error](https://github.com/iomad/iomad/assets/137098267/dcf8bbaa-c991-41b7-aef3-fc23e1a3f21f)

The reason is that the "name" field in the database is varchar(50). This means that if you try to insert 51 or more characters into the name field, Moodle will throw a database error.

This can be avoid by better UI text which will warn the user to only use 50 characters max. We suggest something like this:

![ui tweak](https://github.com/iomad/iomad/assets/137098267/66da06ad-96ce-4b8f-8a2d-172faac6065c)

/blocks/iomad_commerce/edit_course_shopsettings_form.php?shopsettingsid=4&default=0

**Comment by OliverOPI** [comment]
Edits required:

/blocks/iomad_commerce/classes/forms/product_edit_form.php
change line 81 to:

`$mform->addElement('text', 'name', get_string('name', 'block_iomad_commerce'));`

The old line 81 was:
`$mform->addElement('text', 'name', get_string('name'));`

The problem with line 81 as it was before the edit is that it's missing the string component, which means it's not echoing out the string correctly and preventing a new string from being manually entered in the language pack. Changing this here makes proper use of the string declaration in /lang/en.


Update the default string in /blocks/iomad_commerce/lang/en/

Change line 116 to:
`$string['name'] = 'Name (50 chars max)';`

The old line 116 was:
`$string['name'] = 'Name';`

**Comment by turf212** [comment]
It would be better if i;ts increased to match the max value to match the course full name instead - as that probably makes more sense.
